### PR TITLE
Added kowl deployment and kafka log retention setting

### PR DESCRIPTION
**Pull request by paulbourelly999**
# PR Details
## Description

Includes docker-compose configurations for deploying kowl, an open source software to monitor kafka traffic. Also includes environment variables to set kafka log retention time. This is necessary to allow for topic message traffic collection. Current setting is for 24 hours allowing for daily collection of log files.

Logfiles can be collected : 
```
docker exec -it kafka bashrc
kafka-console-consumer.sh --topic <topic to log> --from-beginning --bootstrap-server localhost:9092 > logfile_name.log
exit
docker cp kafka:/logfile_name.log .
```

## Motivation and Context
Log message traffic to collect performance data and calculate metrics

## How Has This Been Tested?
Locally tested. Steps to test this include:

Uncomment kowl deployment from docker-compose
```
docker-compose up -d scheduling_service
docker-compose up -d kowl
python scripts/<message sending scripts>
```
navigate to `locahost:8083` in browser to access kowl UI
navigate to topics tab, select appropriate topic and ensure there is traffic (check kowl functionality)

'''
docker exec -it kafka bashrc
kafka-console-consumer.sh --topic <topic to log> --from-beginning --bootstrap-server localhost:9092 > logfile_name.log
exit
docker cp kafka:/logfile_name.log .
'''
open logfile_name.log and confirm all messages sent are present. (Check logfile collection)

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ x ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ x ] My change requires a change to the documentation.
- [ x ] I have updated the documentation accordingly.
- [ x ] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -100,6 +119,8 @@ services:
     build: 
       context: .
       dockerfile: intersection_model/Dockerfile
+    depends_on:
```
Comment (dan-du-car) [inline docker-compose.yml:122]: Intersection model is not sub/pub anything. I do not think it depends on kafka

File: docker-compose.yml [hunk 2]
```diff
@@ -15,8 +17,24 @@ services:
       KAFKA_ADVERTISED_HOST_NAME:  ${DOCKER_HOST_IP}
       KAFKA_CREATE_TOPICS: "v2xhub_in:1:1,v2xhub_out:1:1,v2xhub_scheduling_plan_sub:1:1,v2xhub_bsm_in:1:1,v2xhub_mobility_operation_in:1:1,v2xhub_mobility_path_in:1:1,vehicle_status_intent_output:1:1"
       KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
+      KAFKA_LOG_RETENTION_HOURS: 24
```
Comment (dan-du-car) [inline docker-compose.yml:20]: What is default retention time?

File: docker-compose.yml [hunk 3]
```diff
@@ -15,8 +17,24 @@ services:
...
     volumes:
       - /var/run/docker.sock:/var/run/docker.sock
+##############################################################
+# Kafka Owl for monitoring kafka traffic. UI can be accessed on 
+# configurable port set in performance/kowl.yaml
```
Comment (dan-du-car) [inline docker-compose.yml:25]: Is not needed?

File: performance/kowl.yaml [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+# Kowl considers config options from three sources:
+# Flags, YAML config and env variables - in that order.
+# Accordingly env variables and YAML configs can overwrite input
+# that is set via flags.
+#
+# --- YAML
+# If you want to configure Kowl using a YAML file, specify the path
+# either by setting the flag -config.filepath or by providing
+# the path via the env variable CONFIG_FILEPATH.
+#
```
Comment (dan-du-car) [inline performance/kowl.yaml:11]: Is not needed?

**Comment by dan-du-car** [comment]
This topic subscribe command does not have the timestamp when the message is produced or received in localhost time

File: docker-compose.yml (hunk 2 above)
Comment (paulbourelly999) [reply docker-compose.yml:20]: 128 hours or seven days

File: docker-compose.yml (hunk 3 above)
Comment (paulbourelly999) [reply docker-compose.yml:25]: This is a UI that monitors kafka traffic. It is not necessary for deployment but can be used to see message traffic live and do debugging tasks like filter messages based on json fields or something.

File: performance/kowl.yaml (hunk 4 above)
Comment (paulbourelly999) [reply performance/kowl.yaml:11]: See above

File: docker-compose.yml (hunk 1 above)
Comment (paulbourelly999) [reply docker-compose.yml:122]: Good point. At the very least the messaging_services should depend on the intersection model. I will make that change now.

Comment (paulbourelly999) [reply docker-compose.yml:122]: Actually that is wrong too right. Ok let me correct this.

**Comment by paulbourelly999** [comment]
> This topic subscribe command does not have the timestamp when the message is produced or received in localhost time

I am not sure I know what you are referring to.

**Review by snallamothu — APPROVED** [review]

**Pull request merged**